File: Pythonleetcode/Leetcode1137.py
class Solution(object):
    def tribonacci(self, n):
        dp=[0]*(n+1)
        dp[0:3]=[0,1,1]
        if n<3:
            return dp[n]
        else:
            for i in range(3,n+1):
                dp[i]=dp[i-3]+dp[i-2]+dp[i-1]
        return dp[-1]



print(Solution().tribonacci(4))
print(Solution().tribonacci(25))

File: CppPractice/CppLeetcode/Leetcode1137.cpp
#include<vector>
using namespace std;

class Solution {
public:
    int tribonacci(int n) 
    {
        vector<int> dp={0,1,1};

        if (n<3)
        {
            return dp[n];
        }
        int cur=3;
        while (cur<=n)
        {
            dp.push_back(dp[cur-1]+dp[cur-2]+dp[cur-3]);
            cur++;
        }
        return dp.back();
    }
};